### PR TITLE
F 30480 custom default publication of documentcategories

### PR DESCRIPTION
**Pull request by martendemos**
**Ticket:** https://yaits.demos-deutschland.de/T30480

To cover different 
In order to reduce the effort on importing documents (ZIP-archive), it should be possible for the user to choose if all containing elements should be initially published or not.

### How to review/test
Import a ZIP-archive in the document section. After importing the files, you should see a checkbox on the overview where you can choose to publish the categories automatically or not. 
All contained elements of the imported zip-archive should be in the selected state.

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

**Comment by martendemos** [comment]
still pushing ....

File: templates/bundles/DemosPlanDocumentBundle/DemosPlanDocument/elements_admin_import.html.twig [hunk 1]
```diff
@@ -11,6 +11,16 @@
         </p>
 
         {% include "@DemosPlanDocument/DemosPlanDocument/elements_admin_import_entry.html.twig" %}
+        {% if hasPermission('feature_auto_switch_element_state') %}
+            <div class="u-mt-0_75">
+                <dp-checkbox
+                    id="r_publish_categories"
+                    checked
+                    :label="{ text: Translator.trans('element.state.autoswitch', { state: Translator.trans('published') }) }"
```
Comment (spiess-demos) [inline templates/bundles/DemosPlanDocumentBundle/DemosPlanDocument/elements_admin_import.html.twig:19]: I find it confusing to (re-)use the already-existing translation key here. `element.state.autoswitch` carries a distinct meaning which does not apply to the current case. I'd rather suggest to add a new translation key `element.state.publish` or the like. To also differentiate what actually happens, i'd also suggest to chose " Kategorie  sofort veröffentlichen".

File: translations/messages+intl-icu.de.yml [hunk 2]
```diff
@@ -889,7 +889,7 @@ element.assigned: "Bezug zu Planungsdokument"
 element.delete: "Element entfernen"
 element.notassigned: "Es ist kein Bezug zu einem Planungsdokument festgelegt."
 element.selected: "Ausgewähltes Element"
-element.state.autoswitch: "Status der Kategorie automatisch auf \"{state}\" umstellen"
+element.state.autoswitch: "Status der Kategorie(n) automatisch auf \"{state}\" umstellen"
```
Comment (spiess-demos) [inline translations/messages+intl-icu.de.yml:892]: See above comment. I'd revert the change to `element.state.autoswitch` and add a new key instead.

**Review by spiess-demos — CHANGES_REQUESTED** [review]

**Comment by gruenbergerdemos** [comment]
Concerns done in [8b5cd10](https://github.com/demos-europe/demosplan-core/pull/546/commits/8b5cd10fbb68d1a3be5710ebab9c5a716a169b6f)

File: templates/bundles/DemosPlanDocumentBundle/DemosPlanDocument/elements_admin_import.html.twig [hunk 3]
```diff
@@ -11,6 +11,16 @@
         </p>
 
         {% include "@DemosPlanDocument/DemosPlanDocument/elements_admin_import_entry.html.twig" %}
+        
+        {% if hasPermission('feature_auto_switch_element_state') %}
+            <dp-checkbox
+                class="u-mt-0_75"
+                id="r_publish_categories"
+                checked
+                :label="{ text: Translator.trans('element.state.publish', { state: Translator.trans('published') }) }"
```
Comment (spiess-demos) [inline templates/bundles/DemosPlanDocumentBundle/DemosPlanDocument/elements_admin_import.html.twig:20]:
```suggestion
                :label="{ text: Translator.trans('element.state.publish') }"
```

**Review by spiess-demos — DISMISSED** [review]
one last suggestion, otherwise lgtm!

**Review by spiess-demos — APPROVED** [review]
FE lgtm (code review).

**Review by mussbach — APPROVED** [review]
lgtm, code review only. Perhaps you could tackle the php 8 deprecation

File: demosplan/DemosPlanDocumentBundle/Logic/DocumentHandler.php [hunk 4]
```diff
@@ -174,7 +188,16 @@ public function getParaDocumentAdminList(string $procedureId, string $elementId)
...
-    protected function saveElementsFromDirArray($entries, $elementId, $sessionId, $procedure, $request, $sessionElementImportList, $category = null, array &$errorReport)
+    protected function saveElementsFromDirArray(
+        $entries,
+        $elementId,
+        $sessionId,
+        $procedure,
+        $request,
+        $sessionElementImportList,
+        $category = null,
+        array &$errorReport
```
[the remark this answers is not in the document]
Comment (MoritzMandler) [reply demosplan/DemosPlanDocumentBundle/Logic/DocumentHandler.php:199]: please change the order of these parameters as review-dog points out.

**Review by MoritzMandler — CHANGES_REQUESTED** [review]
Just the one review-dog parameter-reorder thing...
appart from that code lgtm and worked well when tested

**Review by MoritzMandler — APPROVED** [review]
Thank you - lgtm

**Pull request merged**